### PR TITLE
[11.0][IMP] change date format

### PR DESCRIPTION
**Pull request by ghost**
Backport

### CODE REVIEW

File: cms_form/models/widgets.py [hunk 1]
```diff
@@ -262,19 +263,49 @@ class DateWidget(models.AbstractModel):
     _inherit = 'cms.form.widget.mixin'
     _w_template = 'cms_form.field_widget_date'
 
+    w_data_format = "yy-mm-dd"  # jquery.ui data widget format
+    w_form_date_format = "%Y-%m-%d"  # python date format
+    w_data_date_format = "%Y-%m-%d"  # python date format
+    w_placeholder = "YYYY-MM-DD"
+
+    def data_to_form_date_format(self, date_str):
```
Comment (simahawk) [inline cms_form/models/widgets.py:271]: don't need this

File: cms_form/models/widgets.py [hunk 2]
```diff
@@ -262,19 +263,49 @@ class DateWidget(models.AbstractModel):
...
+        """ Convert from w_data_date_format to w_form_date_format """
+        date = datetime.strptime(date_str, self.w_data_date_format)
+        return date.strftime(self.w_form_date_format)
+
+    def form_to_data_date_format(self, date_str):
```
Comment (simahawk) [inline cms_form/models/widgets.py:276]: don't need this

File: cms_form/models/widgets.py [hunk 3]
```diff
@@ -262,19 +263,49 @@ class DateWidget(models.AbstractModel):
...
+        """ Convert from w_form_date_format to w_data_date_format """
+        date = datetime.strptime(date_str, self.w_form_date_format)
+        return date.strftime(self.w_data_date_format)
+
+    def is_ui_date_format(self, date_str):
```
Comment (simahawk) [inline cms_form/models/widgets.py:281]: don't need this

File: cms_form/models/widgets.py [hunk 4]
```diff
@@ -262,19 +263,49 @@ class DateWidget(models.AbstractModel):
...
+        except ValueError:
+            is_ui_format = False
+        return is_ui_format
+
     def widget_init(self, form, fname, field, **kw):
         widget = super().widget_init(form, fname, field, **kw)
         if 'defaultToday' not in widget.w_data:
             # set today's date by default
             widget.w_data['defaultToday'] = True
+        if '_default' in field and \
```
Comment (simahawk) [inline cms_form/models/widgets.py:294]: don't need this, if you have the default is going to be always the one from the server

File: cms_form/models/widgets.py [hunk 5]
```diff
@@ -262,19 +263,49 @@ class DateWidget(models.AbstractModel):
...
 
     def w_extract(self, **req_values):
         value = super().w_extract(**req_values)
         return self.form_to_date(value, **req_values)
 
     def form_to_date(self, value, **req_values):
-        return utils.safe_to_date(value)
+        value = utils.safe_to_date(value)
+        if value:
+            value = self.form_to_data_date_format(value)
```
Comment (simahawk) [inline cms_form/models/widgets.py:307]:
```python
try:
    dt = datetime.strptime(date_str, self.w_form_date_format)
except:
    logger.debug('Bad request value for %s. Expected format is "%s" % (self.w_fname, self.w_form_date_format)
    return False
value = fields.Date.to_string(dt)
```

File: cms_form/models/widgets.py (hunk 1 above)
Comment (simahawk) [reply cms_form/models/widgets.py:271]: you have to override `w_load` to convert value from server format to w_form_date_format

**Review by Tonow-c2c — APPROVED** [review]
LGTM Just code review

File: cms_form/models/widgets.py [hunk 6]
```diff
@@ -262,19 +263,49 @@ class DateWidget(models.AbstractModel):
     _inherit = 'cms.form.widget.mixin'
     _w_template = 'cms_form.field_widget_date'
 
+    w_data_format = "yy-mm-dd"  # jquery.ui data widget format
+    w_form_date_format = "%Y-%m-%d"  # python date format
+    w_data_date_format = "%Y-%m-%d"  # python date format
+    w_placeholder = "YYYY-MM-DD"
```
Comment (simahawk) [inline cms_form/models/widgets.py:269]: placeholder should be somehow tied to widget format

**Review by simahawk — CHANGES_REQUESTED** [review]

**Comment by simahawk** [comment]
done properly here https://github.com/OCA/website-cms/pull/99

**Pull request closed without merging**